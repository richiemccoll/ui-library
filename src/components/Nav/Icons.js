import React from 'react';

export const Icon = ({ isExpanded }) => {
  return (
    <span>
      {isExpanded
        ? <svg
            id="i-close"
            viewBox="0 0 32 32"
            width="18"
            height="18"
            fill="none"
            stroke="currentcolor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={{'color': 'white'}}
          >
            <path d="M2 30 L30 2 M30 30 L2 2" />
          </svg>
        : <svg
            id="i-menu"
            viewBox="0 0 32 32"
            width="32"
            height="32"
            fill="currentcolor"
            stroke="currentcolor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={{'color': 'white'}}
          >
            <path d="M4 8 L28 8 M4 16 L28 16 M4 24 L28 24" />
          </svg>}
    </span>
  );
};
