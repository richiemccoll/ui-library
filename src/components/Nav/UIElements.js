import styled from 'styled-components';

export const Container = styled.nav`
    background-color: ${props => props.color};
`;

export const NavButton = styled.button`
    background-color: ${props => props.color};
    border: none;
    margin: 8px;
`;

export const NavItems = styled.ul`
    list-style: none;
    padding: 0;
`;

export const Item = styled.li`
    text-align: center;
    text-transform: uppercase;
    color: white;
    padding: 8px;

    a {
        color: white;
    }
`;

export const Title = styled.span`
    align-self: center;
    text-transform: uppercase;
    padding-left: 16px;
`;