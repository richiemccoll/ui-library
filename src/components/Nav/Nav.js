import React from "react";
import PropTypes from "prop-types";
import { Container, NavItems, Item, NavButton } from "./UIElements";
import { Icon } from "./Icons";

const Button = ({ color, isExpanded, showMenu }) =>
  <NavButton color={color} aria-expanded={isExpanded} onClick={showMenu}>
    <Icon isExpanded={isExpanded} />
  </NavButton>;

Button.propTypes = {
  color: PropTypes.string,
  isExpanded: PropTypes.bool.isRequired,
  showMenu: PropTypes.func.isRequired
};

const NavLinks = ({ links }) => {
  return (
    <NavItems>
      {links.map(link => {
        return (
          <Item key={Math.random().toString(36)}>
            <a href={link}>
              {link}
            </a>
          </Item>
        );
      })}
    </NavItems>
  );
};

NavLinks.propTypes = {
    links: PropTypes.arrayOf(PropTypes.string)
}

class Nav extends React.Component {
  state = {
    _ariaExpanded: false,
    _menuHidden: true
  };

  toggle = state => ({
    _menuHidden: !state._menuHidden,
    _ariaExpanded: !state._ariaExpanded
  });

  displayMenuItems = () => this.setState(this.toggle);

  render() {
    return (
      <Container color={this.props.color} aria-label="site-navigation">
        <Button
          color={this.props.color}
          isExpanded={this.state._ariaExpanded}
          showMenu={this.displayMenuItems}
        />
        {this.state._menuHidden
          ? <ul hidden />
          : <NavLinks links={this.props.menuItems} />}
      </Container>
    );
  }
}

export default Nav;

Nav.propTypes = {
    title: PropTypes.string,
    color: PropTypes.string.isRequired,
    menuItems: PropTypes.arrayOf(PropTypes.string)
}
