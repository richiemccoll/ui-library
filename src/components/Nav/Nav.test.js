import React from "react";
import renderer from "react-test-renderer";
import Nav from "./Nav";

describe("Nav Component", () => {
  it("should render correctly", () => {
    const component = renderer.create(
      <Nav title="Navbar" color="#2196F3" menuItems={["Home", "About"]} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
