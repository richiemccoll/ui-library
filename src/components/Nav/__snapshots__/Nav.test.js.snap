// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component should render correctly 1`] = `
<nav
  aria-label="site-navigation"
  className="sc-bdVaJa hyZMgA"
  color="#2196F3"
>
  <button
    aria-expanded={false}
    className="sc-bwzfXH iXNRId"
    color="#2196F3"
    onClick={[Function]}
  >
    <span>
      <svg
        fill="currentcolor"
        height="32"
        id="i-menu"
        stroke="currentcolor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 32 32"
        width="32"
      >
        <path
          d="M4 8 L28 8 M4 16 L28 16 M4 24 L28 24"
        />
      </svg>
    </span>
  </button>
  <ul
    hidden={true}
  />
</nav>
`;
